### PR TITLE
[JENKINS-34952] [WiP] Draw item categories

### PR DESCRIPTION
**Pull request by recena**
[JENKINS-34952](https://issues.jenkins-ci.org/browse/JENKINS-34952)
- [x] New helper methods to handle category data structure
- [x] New CSS for flat categories
- [ ] New render method for category listing

@reviewbybees
@jenkinsci/code-reviewers

### CODE REVIEW

**Comment by lanwen** [comment]
How it should be look like?

File: war/src/main/js/add-item.js [hunk 1]
```diff
@@ -1,6 +1,12 @@
 // Initialize all modules by requiring them. Also makes sure they get bundled (see gulpfile.js).
 var $ = require('jquery-detached').getJQuery();
 
+// Minimum number of item by category to draw the category
+var MIN_ITEM = 6;
```
Comment (oleg-nenashev) [inline war/src/main/js/add-item.js]: This is a pretty big number IMHO. There are not so many Item types in Jenkins, so would be useful to know reason of such value

File: war/src/main/js/add-item.js [hunk 2]
```diff
@@ -300,5 +334,7 @@ $.when(getItems()).done(function(data) {
 
     // Do sticky the form buttons
     doSticky();
+
+    console.log(isDrawableCategories(data.categories));
```
Comment (oleg-nenashev) [inline war/src/main/js/add-item.js]: Is it a development-time message? Does it need removal?

**Comment by oleg-nenashev** [comment]
Generally looks good to me (I've seen the layout at GSoC mid-term presentations)

**Comment by recena** [comment]
@oleg-nenashev this is in progress. I hope to finish it this week.

**Comment by samatdav** [comment]
@recena Could you say when you expect to finish this PR?

**Comment by recena** [comment]
@samatdav No but I hope soon.

**Comment by daniel-beck** [comment]
@recena Any plans to finish this? It's been more than 1.5 years.

**Pull request closed without merging**